# BioMeDS Group Onboarding Information

## Mission Statement

Our mission is to advance the field of (bio)medical research through the development and application of cutting-edge (bio)informatic tools. We are dedicated to unlocking the full potential of biomedical data by creating innovative methods and tools that extract valuable insights and enhance the understanding of complex biological systems.

Our primary objective is to empower researchers and practitioners worldwide, enabling them to harness the full value of their data. By providing user-friendly and powerful (bio)informatic solutions, we strive to facilitate more informed decision-making and accelerate scientific discoveries. Through collaborative efforts with experts in diverse domains, we aim to foster real-world utility and impact in the field of biomedicine.

As advocates of Open Science, we are committed to upholding the highest standards of transparency, reproducibility, and data sharing. We actively promote the dissemination of our methods and tools through publications, ensuring accessibility and facilitating their widespread adoption. We are convinced that openness and collaboration do more to advance science than competition.

BioMedical Data Science also aims to provide high-quality teaching and training to students, nurturing the next generation of (bio)medical data scientists. We believe in equipping students with the necessary skills and knowledge to excel in this rapidly evolving field. Through mentorship, workshops, and educational resources, we strive to cultivate a learning environment that fosters intellectual curiosity, critical thinking, and practical expertise.


## Expectations

**Your role:** We expect that you will take primary responsibility for the success of your research project and career development.
As a member of the lab, you are expected to participate fully in the team.
In general, lab members are free to schedule their working hours according to their own preferences.
Lab members can also freely mix attendance in the lab and home/mobile office.
Participation in scheduled meetings and events of the group and institute is generally expected.
Every member is expected to deliver high quality work, independent of when and where the work is being done.

**Markus's role:** Markus's goal is to facilitate your success as well as that of your project.

**Deadlines:** 
Abstracts for meetings must be shared with all co-authors at least one week prior to the deadline for submission.
Failure to abide by this guideline will result in missing whatever the opportunity in question is.

**Code of Conduct:** All members of the lab, along with visitors, are expected to agree with this code of conduct.
We will enforce this code.
We expect cooperation from all members to help ensuring a safe environment for everybody.
The lab is dedicated to providing a harassment-free experience for everyone, regardless of gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, or religion (or lack thereof).
We do not tolerate harassment of lab members in any form.
Sexual language and imagery is generally not appropriate for any lab venue, including lab meetings, presentations, or discussions.
However, do note that we work on biological matters so work-related discussions of e.g. animal reproduction are appropriate.
Harassment includes offensive verbal comments related to gender, gender identity and expression, age, sexual orientation, disability, physical appearance, body size, race, religion, sexual images in public spaces, deliberate intimidation, stalking, following, harassing photography or recording, sustained disruption of talks or other events, inappropriate physical contact, and unwelcome sexual attention.
Members asked to stop any harassing behavior are expected to comply immediately.

If you are being harassed, notice that someone else is being harassed, or have any other concerns, please contact Markus Ankenbrand immediately. If Markus is the cause of your concern, Prof. Dr. Sabine Fischer is a good informal point of contact; she does not work for Markus and has agreed to mediate.
For additional resolution paths, please see the University of Würzburg conflict management office (https://www.uni-wuerzburg.de/beschaeftigte/konfliktmanagement/anlaufstellen/).
The code of conduct section is licensed under a Creative Commons Attribution 3.0 Unported License. <http://2012.jsconf.us/#/about> & The Ada Initiative.
Please help by translating or improving: <http://github.com/leftlogic/confcodeofconduct.com>.

We expect members to follow these guidelines at any lab-related event.

**Authorship:** Our lab follows the [ICMJE's Uniform Requirements for Manuscripts Submitted to Biomedical Journals](https://www.icmje.org/recommendations/browse/roles-and-responsibilities/defining-the-role-of-authors-and-contributors.html) defintions of the roles of authors and contributors to our manuscripts.

**Ethics:** We expect lab members to be honest in scientific communications both within and outside the lab.
We expect that lab members will design experiments in a manner that minimizes both bias and self deception.
We expect that lab members will keep agreements, be careful, and share their code and results openly with the scientific community.
We expect that credit will be given where credit is due, including in scientific writing.
Plagiarism is not tolerated.
While a full enumeration of ethical considerations is outside of the scope of this document, CU provides a [document](https://www.cu.edu/ope/aps/2027) that we recommend.

In addition, please don't hesitate to raise any questions or concerns that you have at any point with Markus.

**Say 'no':** While we foster collaborations and aim to help our colleagues with any requests, time is an essential and limited resource. Therefore, it is necessary to carefully evaluate each new commitment. Declining (good) opportunities for collaboration or requests for support is a hard thing to do. But in order to make progress in existing projects, while keeping our high standards, it is inevitable. If you have to say 'no', always do so in a curteous but clear manner. Don't leave requests unanswered.

## Communication

### General

**Mattermost:** We use mattermost for rapid communication within the lab.
If you plan on sending an e-mail to someone within the lab, try a mattermost message instead.
This helps to keep communications in one place.
There are many channels on our lab's slack; however, it is recommended that newcomers join the following channels: `biomeds`, `town-square`, `lunch`, `cctb-seminar`.

**Social Media:** Lab members are encouraged to communicate through public social media.
If a lab member chooses to do so on an account that notes an affiliation with the lab, the lab member is expected to follow our code of conduct.
Certain employers may require a disclaimer that the lab member's views do not represent those of the employer.

**Projects:** By the nature of our research, lab members will often have the opportunity to participate in projects managed via private or publicly accessible source code repositories.
In these cases, lab members are expected to: follow the code of conduct; expect that private repositories will be world accessible; and to communicate via the project-specific medium
For example, when one lab member reports an issue on a project on GitHub, it would not be appropriate for another lab member to reply "I'll drop by your desk and show you how to solve that."
It would be most appropriate for the conversation to take place on GitHub issues.

**IP/Openness:** This is handled in accordance with the instructions from our research sponsors and [university guidance](https://www.uni-wuerzburg.de/sft/erfindungen-patente-und-lizenzen-jmu-und-ukw/).
The philosophy of the lab, is to be as open as possible. Restrictions may apply, e.g. due to licensing, privacy, or regulatory concerns.

**Space:** Currently, all lab members, including students get a workspace assigned in the CCTB. You are free to set up this workspace with available hardware from the CCTB. It is your obligation to keep your workplace tidy and clean.

**Calendars:** There is a public Exchange calendar for the group (link in the BioMeDS channel in mattermost, currently read only).
This calendar is used for noting individual availability (i.e. whether you'll be out of office).
It should be used, for example, to note vacations, conference travel, and other workday conflicts.
It also contains group events like lab meetings.

**Accounts:** Lab members are expected to have accounts for the following and be members of the specified (organizations) if applicable:

+ GitHub (BioMeDS)
+ Physics GitLab
+ Mattermost (BioMeDS)

### Meetings

**Lab Meeting:** 
- Lab meetings are scheduled for one microcentury on Wednesdays 3pm. All members of the lab are expected to attend if possible. Meetings are expected to be a supportive environment for learning, constructive criticism, help, and scientific discussions. 

   **Meeting Agenda:** Topics for the next meeting are collected in [the collaborative note](http://10.87.174.31:3000/ZlhAamyNQ3OOLm1QicPIug). You can add any topic, you would like to discuss with the group. This document is also used to keep minutes of the lab meetings. Beside general organizational topics, and those collected in the note, we always do a REPL (retrospective and planning). In the REPL every member briefly reflects on the work since the last meeting, and lays out the plan until the next meeting. Everyone should identify one specific win and one specific tangible goal. Additionally, we regularly do rotations of project presentations or introductions to tools and techniques (announced in advance).

**Ad-Hoc Meetings:** Beside regular lab meetings, any member can schedule an ad hoc meeting as needed. Ad hoc meetings are meant to help lab members get advice and help on projects, prepare for talks, oral exams, etc.

**Individual Meetings:** We schedule individual meetings, at least once per week. These meetings can be live or via Zoom.
 When you join the lab, we will meet more frequently to kick-off your project.
 Once your project is running, we will transition to weekly individual meetings.
 Please take care of scheduling your individual meetings with Markus, yourself.
 The goal of the weekly meeting is to:

 1.  Discuss challenges
 2.  Plan strategy

## Source Code, Data, and Reproducibility

**Pride:** We expect lab members to sign their code.
To quote from _The Pragmatic Programmer_:

> Craftsmen of an earlier age were proud to sign their work.
You should be, too…
People should see your name on a piece of code and expect it to be solid, well written, tested, and documented.

While some code will be proof-of-concept code, it should be of a form that inspires confidence.

**Language:** We write code in a programming language, suitable for the task. For our analyses we mainly use Python, R, or Julia. Knowing the basic syntax of these languagens allows everyone in the lab to understand analytical code.

**Licensing:** We expect code that we produce to be licensed under a MIT license.
Unless a funding agency requires something different, we'll use this.
If you have questions or concerns about licensing, feel free to raise them in Mattermost.

**Version Control Services:** Our primary version control service is GitHub, and we have a [`BioMeDS` organization](https://github.com/BioMeDS) there.
We expect that lab members will maintain their code in repositories under these team accounts.

**Projects that didn't work:** We expect that repositories will contain failures (e.g. proof-of-concepts that didn't work).
This is ideal.
Being able to find them will make sure we don't make the same failure twice.

**Non-Code Versioning:** Data (or models, or anything else), that is too large to be handled with git, should be managed either through DVC or DataLad.

**Data Management:** For publicly available data, scripts used to download and process these data should be preserved, as should the versions of items used in processing (e.g. probe to gene mappings).
These items should be version controlled.
Where possible, intermediate files of reasonable size can be stored to facilitate re-use, but the process to regenerate these files from publicly available data should be preserved.
When we generate data, they should be stored in a location where they are replicated and uploaded to the relevant database as soon as possible (e.g. GEO for gene expression, SRA for sequencing, Zenodo).

**Reproducibility:** We expect all lab members to maintain code that performs reproducible analyses.
This can be in the form of makefiles, shell scripts, or other automation approaches that allow analyses to be automatically performed.
We expect that these scripts, including those to generate figures in papers generated as a consequence of such analyses, will be included in source control repositories and made publicly available before or concurrent with the submission of preprint (if submitted) or manuscripts.

## Protocol Guidelines

Todo

## Additional Resources

- [CCTB getting started guide](https://github.com/CCTB-UW/guides/blob/main/Beginners.md)

